packages/nodes-base/nodes/Google/YouTube/GenericFunctions.ts

```diff
@@ -63,3 +66,17 @@ export async function googleApiRequestAllItems(
 
 	return returnData;
 }
+
+async function* bufferToChunks(buffer: Buffer) {
+	for (let j = 0; j < buffer.length; j += UPLOAD_CHUNK_SIZE) {
+		yield Buffer.from(buffer.subarray(j, j + UPLOAD_CHUNK_SIZE));
+	}
+}
+
+export function getChunkedFileContent(
```

Comment:
This feels like we are re-implementing streaming.
Maybe we can do something like [this](https://github.com/n8n-io/n8n/compare/proposal-for-15405?expand=1) instead?